### ISSUE TITLE
PromiseKit 1.6.x - Support for tvOS and WatchOS

### ISSUE THREAD

**Issue by mofirouz** [mxcl/PromiseKit#311]
I'm developing a library that is dependant on AFN and PromiseKit. The latest AFN (3.0.0-beta) has support for watchOS and tvOS, and so does PromiseKit 2.x/3.x. 

Is there anyway that we can add support for the said targets in PMK 1.6 please?

Thanks!

**Comment by staminajim** [comment]
I can confirm that by adding 
`s.watchos.deployment_target = '2.0'`
to the pod spec with the v1.6.0 tag that using PromiseKit 1.6 works on watchOS.

I could make a pull request, but I'm not sure how you guys plan to keep maintaining 1.6.x as it looks like it's currently all a linear history in master?

**Comment by mofirouz** [comment]
Thanks @staminajim - It seems like WatchOS is working, but tvOS is still no go.

**Comment by staminajim** [comment]
@mofirouz does it work if you add 
`s.tvos.deployment_target = '9.0'` to the 1.6.0 pod spec?

I don't have any tvOS targets set up to quickly try that.

**Comment by mofirouz** [comment]
Sadly I don't have a target that will enable that as well. I've added support for it in the [library](http://github.com/heroiclabs/heroiclabs-sdk-ios) that I'm building for whenever PromiseKit adds that in.

On 24 February 2016 at 10:05:12, James V (notifications@github.com) wrote:

@mofirouz does it work if you add
s.tvos.deployment_target = '9.0' to the 1.6.0 pod spec?

I don't have any tvOS targets set up to quickly try that.

**Comment by jasonzurita** [comment]
I had a need to use Heroic Labs with tvOS, so I figured I would provide an update:

I locally copied both Heroic Labs and PromiseKit (v1.6), modified the pod spec files to include the tvOS target, and the pods successfully installed!

**Comment by mxcl** [comment]
PR welcome, testing this is the tricky part, can advise how.

**Comment by mofirouz** [comment]
@mxcl I would but I honestly don't know where/which branch the code for the 1.6.x release is - similar to what @staminajim said above. 

If you could advise on the git strategy for this project, I'll open a PR asap.

**Comment by mxcl** [comment]
There is no branch, check out the [tag](https://github.com/mxcl/PromiseKit/releases/tag/1.6.0).

    git checkout -b 1.6.0

**Comment by staminajim** [comment]
@mxcl , but don't we need a PK 1.6.x branch to make the pull request against and do future v1 releases from?  We can't clobber master since that's the v3 tree.

**Comment by staminajim** [comment]
Never mind, I just found the discussion continued in the pull request ;)

**Comment by mofirouz** [comment]
Hi @nathanhosselton thanks for merging the PR in. Could I kindly ask for a new release of the 1.x branch so I can upgrade to it and use it in my code please? Thanks.

**Comment by nathanhosselton** [comment]
@mofirouz release it in what way, exactly?

**Comment by mofirouz** [comment]
@nathanhosselton A tag + cocoapods, please :)

**Comment by nathanhosselton** [comment]
So there are numerous failures occurring when attempting to lint the podspec related to the new inclusion of watchOS and tvOS. Aside from the 1.x version of the podspec file not being written in such a way as to trivially handle additional operating system targets, detailed exceptions are going to have to be declared for most of the subspecs' watchOS and tvOS source files, if not in the source code itself.

For example, CLLocationManager's `startUpdatingLocation` method is not available on watchOS or tvOS. On the master branch we simply don't include the CLLocationManager+PromiseKit source files for watchOS in the podspec (which is really a workaround since the functionality could theoretically be provided for those OSs), so the same could be done for 1.x, but this is just one of many updates that will be required. And again, the 1.x and 3.x podspec files are currently very different, so this unfortunately isn't a case where we could just copy+paste.

All this is to say that pushing 1.7.0 to the pod trunk is going to take longer than expected. I have done minimal work to this end since pushing the latest changes to the legacy-1.x branch, so if you were interested in taking this task on then you could start from there. Otherwise I will chip away at it when I can.

**Comment by nathanhosselton** [comment]
OK, I think I understand what changes need to be made to the podspec file now. I can probably knock this out tonight or tomorrow.

**Comment by nathanhosselton** [comment]
Alright, 1.7.0 should be available via CocoaPods now. I tested it locally with watchOS and tvOS targets and everything seemed to be fine, but let me know if you try it out and have issues.

For now I had to make the general sacrifices that I mentioned above in order to at least get it to lint, so some of the category subspecs are unavailable on watchOS and tvOS, most notably the NSURLConnection category. I've left comments in the podspec file at the subspec lines which can work with tvOS/watchOS but don't currently due to API availability differences, so anyone who is feeling adventurous enough can try to take those TODOs on.
